packages/glibc.rb

```diff
@@ -258,20 +503,82 @@ def self.install
+          system "localedef --prefix=#{CREW_DEST_DIR} -i fr_FR -f UTF-8 fr_FR.UTF-8"
+          system "localedef --prefix=#{CREW_DEST_DIR} -i it_IT -f ISO-8859-1 it_IT"
+          system "localedef --prefix=#{CREW_DEST_DIR} -i it_IT -f UTF-8 it_IT.UTF-8"
+          system "localedef --prefix=#{CREW_DEST_DIR} -i ja_JP -f EUC-JP ja_JP"
+          system "localedef --prefix=#{CREW_DEST_DIR} -i ru_RU -f KOI8-R ru_RU.KOI8-R"
+          system "localedef --prefix=#{CREW_DEST_DIR} -i ru_RU -f UTF-8 ru_RU.UTF-8"
+          system "localedef --prefix=#{CREW_DEST_DIR} -i tr_TR -f UTF-8 tr_TR.UTF-8"
+          system "localedef --prefix=#{CREW_DEST_DIR} -i zh_CN -f GB18030 zh_CN.GB18030"
+        end
+      end
+      if @LIBC_VERSION.to_f >= 2.32
+        system "install -Dt #{CREW_DEST_PREFIX}/bin -m755 build-locale-archive"
+        system 'make', "DESTDIR=#{CREW_DEST_DIR}", 'localedata/install-locales'
+      end
     end
+    # libnsl is provided by perl
+    FileUtils.rm_f Dir.glob("#{CREW_DEST_LIB_PREFIX}/libnsl.so*")
     # Remove libmount.so since it conflicts with the one from util_linux.
-    FileUtils.rm Dir.glob("#{CREW_DEST_LIB_PREFIX}/libmount.so.*")
+    FileUtils.rm Dir.glob("#{CREW_DEST_LIB_PREFIX}/libmount.so*")
+    # threads.h was introduced in glibc 2.28. This is a workaround for
+    # pre-M92 systems.
+    return unless @LIBC_VERSION < '2.28'
+
+    system 'curl -Lf https://github.com/jtsiomb/c11threads/raw/19abeee43272002301ddece2f7d5df37394bb54f/c11threads.h -o threads.h'
+    unless Digest::SHA256.hexdigest(File.read('threads.h')) == 'c945fd352449174d3b6107c715b622206ebb81694ac23239637439d78e33ee5a'
+      abort 'Checksum mismatch. :/ Try again.'.lightred
+    end
+    FileUtils.cp 'threads.h', "#{CREW_DEST_PREFIX}/include/"
   end
 
-  # def self.check
-  # Dir.chdir 'glibc_build' do
-  # system "make -k -j#{CREW_NPROC} check"
-  # end
-  # end
+  def self.check
+    # /usr/local/bin/nm: /usr/local/tmp/crew/glibc.20211222010048.dir/glibc-2.23/glibc_build/elf/../string/rtld-memcpy-sse2-unaligned.os: no symbols
+    # /usr/local/bin/nm: /usr/local/tmp/crew/glibc.20211222010048.dir/glibc-2.23/glibc_build/elf/../string/rtld-memmove-sse2-unaligned.os: no symbols
+    # make[2]: Target 'tests' not remade because of errors.
+    # rm /usr/local/tmp/crew/glibc.20211222010048.dir/glibc-2.23/glibc_build/libc.dynsym /usr/local/tmp/crew/glibc.20211222010048.dir/glibc-2.23/glibc_build/elf/ld.dynsym
+    # make[2]: Leaving directory '/usr/local/tmp/crew/glibc.20211222010048.dir/glibc-2.23/elf'
+    # make[1]: *** [Makefile:214: elf/tests] Error 2
+    # make[1]: Target 'check' not remade because of errors.
+    # make[1]: Leaving directory '/usr/local/tmp/crew/glibc.20211222010048.dir/glibc-2.23'
+    # make: *** [Makefile:9: check] Error 2
+    # return if ARCH == 'i686'
+
+    # Dir.chdir 'glibc_build' do
+    # system "make -k -j#{CREW_NPROC} check || true"
+    # end
```

Comment:
I take it the tests fail?  Can we add a note to explain the error and why it was commented?

Reply:
Sure

Reply:
Comments added.